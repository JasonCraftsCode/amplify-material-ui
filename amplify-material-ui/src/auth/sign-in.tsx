import * as React from 'react';
import {
    Button,
    Grid,
    Link,
    TextField,
    makeStyles,
    createStyles,
    Theme,
} from '@material-ui/core';
import Auth from '@aws-amplify/auth';
import { ConsoleLogger as Logger, I18n, JS } from '@aws-amplify/core';

import { AuthProps } from './auth-props';
import { FormSection, SectionHeader, SectionBody, SectionFooter } from '../ui';
import { useForm } from '../hooks';

const logger = new Logger('SignIn');

export interface SignInProps extends AuthProps {
    validationData?: { [key: string]: string };
}

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        form: {
            width: '100%', // Fix IE 11 issue.
            marginTop: theme.spacing(1),
        },
        submit: {
            margin: theme.spacing(3, 0, 2),
        },
    }),
);

export const SignIn: React.FC<SignInProps> = props => {
    const { authState, onStateChange, validationData } = props;
    const classes = useStyles();

    const checkContact = async (user: any) => {
        if (!Auth || typeof Auth.verifiedContact !== 'function') {
            throw new Error(
                'No Auth module found, please ensure @aws-amplify/auth is imported',
            );
        }
        const data = await Auth.verifiedContact(user);
        if (!JS.isEmpty(data.verified)) {
            onStateChange('signedIn', user);
        } else {
            user = Object.assign(user, data);
            onStateChange('verifyContact', user);
        }
    };

    const signIn = async (inputs: any) => {
        if (!Auth || typeof Auth.signIn !== 'function') {
            throw new Error(
                'No Auth module found, please ensure @aws-amplify/auth is imported',
            );
        }

        const { username, password } = inputs;

        try {
            const user = await Auth.signIn({
                username,
                password,
                validationData,
            });
            logger.debug(user);
            if (
                user.challengeName === 'SMS_MFA' ||
                user.challengeName === 'SOFTWARE_TOKEN_MFA'
            ) {
                logger.debug('confirm user with ' + user.challengeName);
                onStateChange('confirmSignIn', user);
            } else if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
                logger.debug('require new password', user.challengeParam);
                onStateChange('requireNewPassword', user);
            } else if (user.challengeName === 'MFA_SETUP') {
                logger.debug('TOTP setup', user.challengeParam);
                onStateChange('TOTPSetup', user);
            } else {
                checkContact(user);
            }
        } catch (err) {
            if (err.code === 'UserNotConfirmedException') {
                logger.debug('the user is not confirmed');
                onStateChange('confirmSignUp', { username });
            } else if (err.code === 'PasswordResetRequiredException') {
                logger.debug('the user requires a new password');
                onStateChange('forgotPassword', { username });
            } else {
                //this.error(err);
                console.log(err);
            }
        } finally {
            //this.setState({ loading: false });
        }
        //onStateChange('signedIn', null); //TODO
    };

    const { inputs, handleInputChange, handleSubmit } = useForm(signIn, {
        username: '',
        password: '',
        token: '',
    });

    if (!['signIn', 'signedOut', 'signedUp'].includes(authState)) {
        return null;
    }

    return (
        <FormSection>
            <SectionHeader>{I18n.get('Sign in to your account')}</SectionHeader>
            <form
                onSubmit={handleSubmit}
                className={classes.form}
                noValidate
                data-testid="signInForm"
            >
                <SectionBody>
                    <TextField
                        variant="outlined"
                        margin="normal"
                        required
                        fullWidth
                        id="username"
                        label="Username"
                        name="username"
                        autoFocus
                        onChange={handleInputChange}
                        value={inputs.username}
                    />
                    <TextField
                        variant="outlined"
                        margin="normal"
                        required
                        fullWidth
                        name="password"
                        label="Password"
                        type="password"
                        id="password"
                        autoComplete="current-password"
                        onChange={handleInputChange}
                        value={inputs.password}
                    />
                </SectionBody>
                <SectionFooter>
                    <Button
                        type="submit"
                        fullWidth
                        variant="contained"
                        color="primary"
                        className={classes.submit}
                        data-testid="signInSubmit"
                    >
                        Sign In
                    </Button>
                    <Grid container>
                        <Grid item xs>
                            <Link
                                href="#"
                                onClick={() =>
                                    onStateChange('forgotPassword', null)
                                }
                                variant="body2"
                            >
                                Forgot password?
                            </Link>
                        </Grid>
                        <Grid item>
                            <Link
                                href="#"
                                onClick={() => onStateChange('signUp', null)}
                                variant="body2"
                            >
                                {"Don't have an account? Sign Up"}
                            </Link>
                        </Grid>
                    </Grid>
                </SectionFooter>
            </form>
        </FormSection>
    );
};
