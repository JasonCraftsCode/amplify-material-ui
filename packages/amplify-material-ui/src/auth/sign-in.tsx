import * as React from 'react';
import invariant from 'tiny-invariant';
import {
    Button,
    Grid,
    makeStyles,
    createStyles,
    Theme,
} from '@material-ui/core';
import Auth from '@aws-amplify/auth';
import { ConsoleLogger as Logger, I18n } from '@aws-amplify/core';
import { Formik, Field, Form } from 'formik';
import { TextField } from 'formik-material-ui';

import { useAuthContext } from './auth-context';
import { useCheckContact } from './use-check-contact';
import { useUsernameField } from './use-username-field';
import { ChangeAuthStateLink } from './change-auth-state-link';
import { FormSection, SectionHeader, SectionBody, SectionFooter } from '../ui';

const logger = new Logger('SignIn');

export const useSignIn = (validationData?: {
    [key: string]: string;
}): ((username: string, password: string) => Promise<void>) => {
    const { onStateChange } = useAuthContext();

    const checkContact = useCheckContact();

    return async (username: string, password: string): Promise<void> => {
        invariant(
            Auth && typeof Auth.signIn === 'function',
            'No Auth module found, please ensure @aws-amplify/auth is imported',
        );

        try {
            const user = await Auth.signIn({
                username,
                password,
                validationData,
            });
            logger.debug(user);
            if (
                user.challengeName === 'SMS_MFA' ||
                user.challengeName === 'SOFTWARE_TOKEN_MFA'
            ) {
                logger.debug('confirm user with ' + user.challengeName);
                onStateChange('confirmSignIn', user);
            } else if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
                logger.debug('require new password', user.challengeParam);
                onStateChange('requireNewPassword', user);
            } else if (user.challengeName === 'MFA_SETUP') {
                logger.debug('TOTP setup', user.challengeParam);
                onStateChange('TOTPSetup', user);
            } else {
                checkContact(user);
            }
        } catch (err) {
            if (err.code === 'UserNotConfirmedException') {
                logger.debug('the user is not confirmed');
                onStateChange('confirmSignUp', { username });
            } else if (err.code === 'PasswordResetRequiredException') {
                logger.debug('the user requires a new password');
                onStateChange('forgotPassword', { username });
            } else {
                //this.error(err);
                console.log(err);
            }
        } finally {
            //this.setState({ loading: false });
        }
        //onStateChange('signedIn', null); //TODO
    };
};

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        form: {
            width: '100%', // Fix IE 11 issue.
            marginTop: theme.spacing(1),
        },
        submit: {
            margin: theme.spacing(3, 0, 2),
        },
    }),
);

export interface SignInProps {
    validationData?: { [key: string]: string };
    hideSignUpLink?: boolean;
    hideForgotPasswordLink?: boolean;
}

export const SignIn: React.FC<SignInProps> = props => {
    const {
        validationData,
        hideSignUpLink = false,
        hideForgotPasswordLink = false,
    } = props;

    const classes = useStyles();
    const signIn = useSignIn(validationData);
    const { usernamefieldName, usernameField } = useUsernameField();

    return (
        <Formik<{ [fieldName: string]: string; password: string }>
            initialValues={{
                [usernamefieldName]: '',
                password: '',
            }}
            onSubmit={async (values, { setSubmitting }): Promise<void> => {
                await signIn(values[usernamefieldName], values['password']);
                setSubmitting(false);
            }}
        >
            {({ handleSubmit, isValid }): React.ReactNode => (
                <FormSection>
                    <SectionHeader>
                        {I18n.get('Sign in to your account')}
                    </SectionHeader>
                    <Form
                        className={classes.form}
                        onSubmit={handleSubmit}
                        data-testid="signInForm"
                        //noValidate
                    >
                        <SectionBody>
                            {usernameField}
                            <Field
                                variant="outlined"
                                margin="normal"
                                required
                                fullWidth
                                name="password"
                                label={I18n.get('Password')}
                                type="password"
                                id="password"
                                autoComplete="current-password"
                                component={TextField}
                            />
                        </SectionBody>
                        <SectionFooter>
                            <Button
                                type="submit"
                                disabled={!isValid}
                                fullWidth
                                variant="contained"
                                color="primary"
                                className={classes.submit}
                                data-testid="signInSubmit"
                            >
                                {I18n.get('Sign In')}
                            </Button>
                            <Grid container>
                                {!hideForgotPasswordLink && (
                                    <Grid item xs>
                                        <ChangeAuthStateLink
                                            label={I18n.get('Reset password')}
                                            newState="forgotPassword"
                                        />
                                    </Grid>
                                )}
                                {!hideSignUpLink && (
                                    <Grid item>
                                        <ChangeAuthStateLink
                                            label={I18n.get('Create account')}
                                            newState="signUp"
                                        />
                                    </Grid>
                                )}
                            </Grid>
                        </SectionFooter>
                    </Form>
                </FormSection>
            )}
        </Formik>
    );
};
