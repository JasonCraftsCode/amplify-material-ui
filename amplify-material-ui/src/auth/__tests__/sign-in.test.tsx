import * as React from 'react';
import Auth from '@aws-amplify/auth';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { SignIn } from '../sign-in';

const acceptedStates = ['signIn', 'signedUp', 'signedOut'];
const deniedStates = [
    'signUp',
    'signedIn',
    'comfirmSignIn',
    'confirmSignUp',
    'forgotPassword',
    'verifyContact',
];

describe('sign-in', () => {
    beforeEach(() => cleanup());

    it('should be rendered correctly in the signIn, signedOut or signedUp authState', () => {
        acceptedStates.forEach(state => {
            const { asFragment } = render(
                <SignIn onStateChange={() => {}} authState={state} />,
            );
            expect(asFragment()).toMatchSnapshot();
        });
    });

    it('should render null if state is unvalid', () => {
        deniedStates.forEach(state => {
            const { asFragment } = render(
                <SignIn onStateChange={() => {}} authState={state} />,
            );
            expect(asFragment()).toMatchSnapshot();
        });
    });

    it('it should change state to requireNewPassword if challengeName equals NEW_PASSWORD_REQUIRED', async () => {
        const handleStateChange = jest.fn();

        jest.spyOn(Auth, 'signIn').mockImplementationOnce((user, password) => {
            return new Promise((res, rej) => {
                res({
                    challengeName: 'NEW_PASSWORD_REQUIRED',
                });
            });
        });

        const { getByTestId, getByLabelText } = render(
            <SignIn onStateChange={handleStateChange} authState="signIn" />,
        );

        const emailInput = getByLabelText('Email Address', {
            exact: false,
            selector: 'input',
        });
        fireEvent.change(emailInput, { target: { value: 'test@test.de' } });

        const passwordInput = getByLabelText('Password', {
            exact: false,
            selector: 'input',
        });
        fireEvent.change(passwordInput, { target: { value: 'Qwertz123!' } });

        fireEvent.click(getByTestId('signInSubmit'));

        await Promise.resolve();

        expect(handleStateChange).toHaveBeenCalledTimes(1);
        expect(handleStateChange).toHaveBeenCalledWith('requireNewPassword', {
            challengeName: 'NEW_PASSWORD_REQUIRED',
        });
    });
});
