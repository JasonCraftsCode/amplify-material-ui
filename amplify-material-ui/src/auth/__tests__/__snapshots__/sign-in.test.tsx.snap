// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-in should be rendered correctly in the signIn, signedOut or signedUp authState 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-41 makeStyles-box-38"
      >
        <div
          class="MuiAvatar-root makeStyles-avatar-39 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity=".87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          Sign in to your account
        </h1>
      </div>
      <form
        class="makeStyles-form-1"
        data-testid="signInForm"
        novalidate=""
      >
        <div
          class="MuiBox-root MuiBox-root-85 makeStyles-box-84"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="email"
            >
              Email Address
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-140"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
            >
              Password
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-140"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-142 makeStyles-box-141"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-submit-2 MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="signInSubmit"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign In
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Forgot password?
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`sign-in should be rendered correctly in the signIn, signedOut or signedUp authState 2`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-282 makeStyles-box-38"
      >
        <div
          class="MuiAvatar-root makeStyles-avatar-39 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity=".87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          Sign in to your account
        </h1>
      </div>
      <form
        class="makeStyles-form-1"
        data-testid="signInForm"
        novalidate=""
      >
        <div
          class="MuiBox-root MuiBox-root-283 makeStyles-box-84"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="email"
            >
              Email Address
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-140"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
            >
              Password
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-140"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-284 makeStyles-box-141"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-submit-2 MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="signInSubmit"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign In
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Forgot password?
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`sign-in should be rendered correctly in the signIn, signedOut or signedUp authState 3`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-285 makeStyles-box-38"
      >
        <div
          class="MuiAvatar-root makeStyles-avatar-39 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0V0z"
              />
              <path
                d="M0 0h24v24H0V0z"
                opacity=".87"
              />
            </g>
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          Sign in to your account
        </h1>
      </div>
      <form
        class="makeStyles-form-1"
        data-testid="signInForm"
        novalidate=""
      >
        <div
          class="MuiBox-root MuiBox-root-286 makeStyles-box-84"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
              data-shrink="true"
              for="email"
            >
              Email Address
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-140"
                  style="width: 0px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
            >
              Password
              <span
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-139 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-140"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-287 makeStyles-box-141"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-submit-2 MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="signInSubmit"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign In
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Forgot password?
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
                href="#"
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`sign-in should render null if state is unvalid 1`] = `<DocumentFragment />`;

exports[`sign-in should render null if state is unvalid 2`] = `<DocumentFragment />`;

exports[`sign-in should render null if state is unvalid 3`] = `<DocumentFragment />`;

exports[`sign-in should render null if state is unvalid 4`] = `<DocumentFragment />`;

exports[`sign-in should render null if state is unvalid 5`] = `<DocumentFragment />`;

exports[`sign-in should render null if state is unvalid 6`] = `<DocumentFragment />`;
